lib/adapter.js

```diff
@@ -1070,10 +1070,26 @@ function Adapter(options) {
         if (state !== undefined) {
             delete task.state;
         }
-        adapterObjects.extendObject(task._id, task, () =>
-            state ?
-                adapterStates.setState(task._id, state, () => setImmediate(extendObjects, tasks, callback)) :
-                setImmediate(extendObjects, tasks, callback));
+        adapterObjects.getObject(task._id, (err, obj) => {
+            if (obj || state !== undefined) {
+                // state will not be newly created, or we have state, thus not set default value
+                adapterObjects.extendObject(task._id, task, () =>
+                    state ?
```

Comment:
This will not work if state is "false" or "Null", or !?

Reply:
I think you also need to check for !== undefined here

Reply:
True, was basically obtaining the existing code